### ISSUE TITLE
Bug: Label_8E unconditionally returns decoded:true / decodeLevel:'full' even when its regex didn't match

### ISSUE THREAD

**Issue by kevinelliott** [airframesio/acars-decoder-typescript#463]
## Summary

`Label_8E.decode` checks `results?.groups` before populating the ETA and arrival-airport fields, but then sets `decoded = true` and `decoder.decodeLevel = 'full'` **unconditionally** — even when the regex didn't match. The plugin therefore reports every 8E message as fully decoded, including ones it produced zero fields for.

Two consequences:

1. **Downstream false positive.** Any caller checking `result.decoded` or `result.decoder.decodeLevel === 'full'` sees success and won't fall back to a "raw" rendering, but `formatted.items` is empty. The user sees a "fully decoded ETA report" with no data.
2. **Loop short-circuit.** `MessageDecoder.decode` (lib/MessageDecoder.ts:193-203) stops trying further plugins on the first `decoded: true`. Today `Label_8E` is the only `8E`-labeled plugin, so this only matters if a second 8E plugin is added later — but the false-success signal itself is the real bug, and the loop-short-circuit risk is a latent footgun for whoever adds the next 8E variant.

## Affected code

`lib/plugins/Label_8E.ts:24-43`

```ts
const regex = /^(?<arrival_icao>\w{4}),(?<arrival_eta>\d{4})$/;
const results = message.text.match(regex);
if (results?.groups) {
  if (options.debug) {
    console.log('Label 8E ETA: groups');
    console.log(results.groups);
  }

  ResultFormatter.eta(
    decodeResult,
    DateTimeUtils.convertHHMMSSToTod(results.groups.arrival_eta),
  );
  ResultFormatter.arrivalAirport(decodeResult, results.groups.arrival_icao);
}

decodeResult.decoded = true;                  // ← runs regardless of match
decodeResult.decoder.decodeLevel = 'full';    // ← runs regardless of match

return decodeResult;
```

For comparison, sibling plugins like `Label_4T_AGFSR`, `Label_2P_FM3`, etc., correctly set `decoded = false` / `decodeLevel = 'none'` in their unknown branches.

## Reproduction

```ts
const decoder = new MessageDecoder();
const r = decoder.decode({ label: '8E', text: 'NOT A VALID 8E' });
// r.decoded === true
// r.decoder.decodeLevel === 'full'
// r.formatted.items.length === 0
// r.decoder.name === 'label-8e'
```

The existing test (`lib/plugins/Label_8E.test.ts`) only covers the happy path (`'EGSS,1618'`), so this code path is uncovered.

## Suggested fix

Move the `decoded`/`decodeLevel` assignments inside the `if (results?.groups)` block, and set the unknown state otherwise — matching the pattern used elsewhere in the codebase:

```ts
if (results?.groups) {
  ResultFormatter.eta(
    decodeResult,
    DateTimeUtils.convertHHMMSSToTod(results.groups.arrival_eta),
  );
  ResultFormatter.arrivalAirport(decodeResult, results.groups.arrival_icao);
  decodeResult.decoded = true;
  decodeResult.decoder.decodeLevel = 'full';
} else {
  ResultFormatter.unknown(decodeResult, message.text);
  decodeResult.decoded = false;
  decodeResult.decoder.decodeLevel = 'none';
}

return decodeResult;
```

## Test coverage

Add a "not a valid 8E" case to `Label_8E.test.ts` that asserts `decoded === false`, `decodeLevel === 'none'`, and `formatted.items.length === 0`.
